proposals/60-clearing-house-batching-service.md

```diff
@@ -0,0 +1,128 @@
+# Filecoin Clearing House / Batching Service protocol
+
+Authors: @rvagg
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/60
+
+## Purpose &amp; impact 
+
+#### Background &amp; intent
+
+_Describe the desired state of the world after this project? Why does that matter?_
+
+* Small deals are hard, and economically unattractive for miners; anecdotally, with current costs, attractive deals are in the order of 16 GiB for standard and 4 GiB for FIL+
+* Many use cases involve parties with _only_ small deals to make and little opportunity to batch their own data
+* Dealing with fr32 padding combined with needing to have deals be power-of-two sized will generally lead to waste on the client side (e.g. [see this table for optimal deal sizes](https://github.com/filecoin-project/filecoin-docs/issues/705#issuecomment-787618483))
+* The deal-making latency is quite large, in the order of 12-24 hours for many deals to be completed and sealed
+* Deals require relationships with miners, the overhead of acquiring and maintaining relationships is a negative for developers building on Filecoin
+* The overhead of managing redundancy, monitoring deal-lifecycle, renewing deals, etc. is a negative for developers building on Filecoin
+
+Batching deals to achieve optimal deal size for miners should enable a better matching of incentives between client and miner. Batching can either be done at the application layer, or it could be done as a service, aggregated across many applications. A batching service **"protocol"** could define a standard API and workflow that could be adopted by layer 1 ecosystem partners and present a common interface to application developers and a simpler set of concerns and rules required to store assets on Filecoin.
+
+This proposal is for developing a protocol and an operational reference implementation.
+
+* The batching service is responsible for making the deals direct with miners. Deals would be large in size, ideally 16 GiB or 32 GiB and would be FIL+.
+* Users of the service only need to establish a relationship with the batching service, not with miners (for storage, retrieval is out of scope for this initial proposal).
+* Users make **micro-deals** via the batching service, depositing FIL with the service in batches to be consumed in small amounts for each deal.
+* Users submit data via an API or IPFS (or other means) and the data is placed in a queue.
+* The batching service aggregates enough data to fit into an optimal deal size, pre-determined for the service, perhaps with a compromise where it will make smaller deals when it hasn't received enough data in a certain number of hours.
```

Comment:
So one complication here (which may vary from service provider to service provider) is that the economics are still sort of weird: 
1) The time guarantee for sealing could leave a sector mostly empty unless the provider has a guaranteed set of deals coming through (or like an average flow rate)
2) The "cost" for paying the diff between the cost of sealing for a miner + the fil collected may be higher for small files. 

I think there's a few arguments that could be made - either that cost is borne by the service provider (provided their making sufficient profit), or it could be borne by the miner (cost of maintaining a relationship with the provider).